repository/fetch-repos.go

```diff
@@ -131,3 +132,310 @@ func getReposByOrg(config *config.GitXargsConfig) ([]*github.Repository, error)
 
 	return allRepos, nil
 }
+
+// getReposBySearch uses GitHub's search API to find repositories matching the given query
+func getReposBySearch(config *config.GitXargsConfig) ([]*github.Repository, error) {
```

Comment:
Can we eliminate this function and call `getReposByIntersection` etc directly from select-repos?